airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -1014,8 +1014,9 @@ def start_sql_job(
         self.log.info("Output: %s", proc.stdout.decode())
         self.log.warning("Stderr: %s", proc.stderr.decode())
         self.log.info("Exit code %d", proc.returncode)
+        stderr_last_20_lines = "\n".join(proc.stderr.decode().strip().splitlines()[-20:])
         if proc.returncode != 0:
-            raise AirflowException(f"Process exit with non-zero exit code. Exit code: {proc.returncode}")
+            raise AirflowException(f"Process exit with non-zero exit code. Exit code: \ {proc.returncode} \n Error Details : {stderr_last_20_lines}")
```

Comment:
I see that we already log the stderr lines in warning level, so maybe we could move this part to the condition and log it in error level if the return code is not equal to 0, and if it is 0 we log it in warning level.
WDYT?
something like:
```python
        if proc.returncode != 0:
            self.log.error("Stderr: %s", proc.stderr.decode())
            self.log.info("Exit code %d", proc.returncode)
            raise AirflowException(f"Process exit with non-zero exit code. Exit code: {proc.returncode}")
       else:
           self.log.warning("Stderr: %s", proc.stderr.decode())
           self.log.info("Exit code %d", proc.returncode)
```

Reply:
Thanks for your reply @hussein-awala ! 
The purpose of displaying the last 20 lines of the code in the exception is so that it is descriptive and one does not have to go through all the logs to figure out the root cause of the issue.
Looks to me that the exception still would not have the error details from above snippet. Please let me know if I misunderstood your approach.